.github/workflows/tests.yml

```diff
@@ -27,4 +27,9 @@ jobs:
           pip install poetry
           poetry install -E "askar bbs"
       - name: Tests
-        run: poetry run pytest
+        run: |
+          poetry run pytest 2>&1 | tee pytest.log
+          if grep -Eq "RuntimeWarning: coroutine .* was never awaited" pytest.log; then
+            echo "Failure: Detected unawaited coroutine warning in pytest output."
+            exit 1
+          fi
```

Comment:
I think it is wise to fail tests on coroutine never awaited errors. I have an inkling that pytest may support this directly? Perhaps it just has a "fail on warning" and "don't fail on warning" and doesn't allow granular failures. Will piping to `tee` preserve the exit code for the pytest invocation? In other words, will the action still fail when there's a test failure and not just on a warning we don't want to see?

Reply:
Ah yes, I thought the same to config pytest to fail on warning, but afaik it only lets you specify the kind of warning (RuntimeWarning / Deprecation warning) as a catch all. So to fail on specific warnings can be solved with this pipe to file and grep approach.

I also wondered if the pytest exit code is preserved -- thought the `2>&1` logic might accommodate that, but not sure. I'll test and refactor as needed 👌

Reply:
@dbluhm  Indeed! It wasn't preserving the exit code ... doh. (Thanks GPT)
Luckily one can use `${PIPESTATUS[0]}`, because PIPESTATUS preserves the exit code of the most recent pipeline.
(Also thanks GPT) :smile: 

Testing fail/success cases now